### ISSUE TITLE
msconvert for Wiff 

### ISSUE THREAD

**Issue by jj-umn** [galaxyproteomics/tools-galaxyp#23]
@jmchilton  @iracooke   I need to add support for converting Wiff files to the msconvert tools.  

ProteoWizard Reader_ABI.cpp checks for the existence of a wiff scan file by appending ".scan"  to the given input wiff file and searching for that filename in the same directory.   

msconvert_wrapper.py as currently coded, doesn't have an option to  copy in the .scan file without also adding it explicitly to the inputs added to the command line.   

Any thoughts on how best to add wiff conversion support?  

If no one has used the wiff datatype as yet, would it be acceptible to add explicit Metadata fields for the .wiff and .wiff.scan files similar to: BowtieIndex in lib/galaxy/datatypes/ngsindex.py or SnpSiftDbNSFP in 
lib/galaxy/datatypes/text.py allowing direct use of ${input.extra_files_path}/${input.metadata.wiff} and  ${input.extra_files_path}/${input.metadata.scan} in command line generation?  This would allow original filenames to be used in the extra_files_path.   

class Wiff( Binary ):
    """Class for wiff files."""
    MetadataElement( name='reference_name', default='ABSCIEX' , desc='Reference Name', readonly=True, visible=True, set_in_upload=True, no_value='ABSCIEX' )
    MetadataElement( name="wiff", default=None, desc="reference_name.wiff", readonly=True, visible=True, no_value=None )
    MetadataElement( name="scan", default=None, desc="reference_name.wiff.scan", readonly=True, visible=True, no_value=None)

**Comment by bgruening** [comment]
@jj-umn are referring to this definition? https://bitbucket.org/iracooke/proteomics-toolshed-datatypes/commits/7f88b4cd778043fe9e8c0dddbe4d785ac3dd3127
I can be that the `.scan` file needs to sit next to the `.wiff` file. So in worst case you need to symlink both files.
Your metadata solution seem to be very clean to me.

**Comment by jj-umn** [comment]
Yes, I am referring to Ira's datatypes.  With current datatype, I would symlink to ${input_name}.wiff and $(input_name}.wiff.scan and would also need to add a "--implicit_inputs" option to msconvert.py that would put the .scan input in the correct dir/name for msconvert without adding it to the msconvert commndline.   Just thought our msconvert/windows experts might have a better alternative.   
( I think that yesterday was my first login to a windows system since 1990.  Damn, I had such a good streak going. )

**Comment by bgruening** [comment]
:tada: hilarious comment :tada:

**Comment by jmchilton** [comment]
@jj-umn I have never done anything like this but this seems like a great approach. Good luck.
